core/amber/src/main/scala/edu/uci/ics/texera/workflow/operators/source/scan/text/FileScanSourceOpExec.scala

```diff
@@ -0,0 +1,68 @@
+package edu.uci.ics.texera.workflow.operators.source.scan.text
+
+import edu.uci.ics.texera.workflow.common.operators.source.SourceOperatorExecutor
+import edu.uci.ics.texera.workflow.common.tuple.Tuple
+import edu.uci.ics.texera.workflow.common.tuple.schema.AttributeTypeUtils
+import org.apache.commons.io.IOUtils.toByteArray
+
+import java.io._
+import java.nio.file.{Files, Paths}
+import java.util.zip.ZipFile
+import scala.jdk.CollectionConverters.{
+  asScalaIteratorConverter,
+  enumerationAsScalaIteratorConverter
+}
+
+class FileScanSourceOpExec private[text] (val desc: FileScanSourceOpDesc)
+    extends SourceOperatorExecutor {
+
+  @throws[IOException]
+  override def produceTexeraTuple(): Iterator[Tuple] = {
+    if (desc.unzip) {
+      val zipReader = new ZipFile(desc.filePath.get)
+      val entries =
+        zipReader.entries().asScala.filterNot(entry => entry.getName.startsWith("__MACOSX"))
+      if (desc.attributeType.isSingle) {
+        entries.map(entry => singleTuple(toByteArray(zipReader.getInputStream(entry))))
+      } else {
+        entries.flatMap(entry => multipleTuple(zipReader.getInputStream(entry)))
+      }
+    } else {
+      if (desc.attributeType.isSingle) {
+        Iterator(singleTuple(Files.readAllBytes(Paths.get(desc.filePath.get))))
+      } else {
+        multipleTuple(new FileInputStream(desc.filePath.get))
+      }
+    }
+  }
```

Comment:
there are too many if-else branches and the code seems duplicating. can we simplify it?

Reply:
Fxied by @Yicong-Huang.